packages/compile-solidity/test/compileWithPragmaAnalysis.js

```diff
@@ -149,6 +144,25 @@ describe("compileWithPragmaAnalysis", function () {
   });
 
   describe("when there is a semver expression error", function () {
+    it("throws an error when a 'direct source' lacks a pragma expression", async function () {
+      try {
+        await compileWithPragmaAnalysis({
+          options: config,
+          paths: [path.join(__dirname, "sources", "badSources", "NoPragma.sol")]
+        });
+        assert.fail("The function should have thrown.");
+      } catch (error) {
+        const expectedMessage = "Could not find a valid pragma expression";
+        if (error.message.includes(expectedMessage)) {
```

Comment:
Rather than using this awkward construct with `return;` (and if you are going to do this I would suggeste eliminating the `"all good"` return value as placing that value there suggests it's going to be used for something and it isn't), I think it would just make sense to invert the conditional and put the failure in the `if`.

Reply:
Yeah I thought about removing that, it sounds like a good idea.